eval/eval.go

```diff
@@ -101,19 +100,23 @@ func (s *State) evalPostfixExpression(node *ast.PostfixExpression) object.Object
 	default:
 		return object.Error{Value: "unknown postfix operator: " + node.Type().String()}
 	}
+	var oerr object.Object
 	switch val := val.(type) {
 	case object.Integer:
-		s.env.Set(id, object.Integer{Value: val.Value + toAdd})
+		oerr = s.env.Set(id, object.Integer{Value: val.Value + toAdd})
 	case object.Float:
-		s.env.Set(id, object.Float{Value: val.Value + float64(toAdd)})
+		oerr = s.env.Set(id, object.Float{Value: val.Value + float64(toAdd)}) // So PI++ fails not silently.
 	default:
 		return object.Error{Value: "can't increment/decrement " + val.Type().String()}
 	}
+	if oerr.Type() == object.ERROR {
+		return oerr
+	}
 	return val
 }
 
 // Doesn't unwrap return - return bubbles up.
-func (s *State) evalInternal(node any) object.Object {
+func (s *State) evalInternal(node any) object.Object { //nolint:funlen // quite a lot of cases.
```

Comment:
Maybe you could add a map[type]func(Node) Object 

Then you create one method per type

So evalInternal method will then simply check if there is a func, then call it

Reply:
What do you think about this?

Reply:
Thanks for the reminder, I had skipped that one earlier

If it worked, we'd need to profile it (good that there is `-profile-cpu` now) but I don't think it'd help because you'd need to cast anyway inside each small function (but try and let me know :) ?)

Note that for other types, like token.Type or object.Type an array lookup would be good, as the type fit into 256 and I had started something like that in #93